crates/services/p2p/src/peer_manager.rs

```diff
@@ -302,6 +300,18 @@ impl NetworkBehaviour for PeerInfoBehaviour {
         cx: &mut Context<'_>,
         params: &mut impl PollParameters,
     ) -> Poll<NetworkBehaviourAction<Self::OutEvent, Self::ConnectionHandler>> {
+        if let Some(event) = self.peer_manager.pending_events.pop_front() {
+            return Poll::Ready(NetworkBehaviourAction::GenerateEvent(event))
+        }
+
+        if self.health_check.poll_tick(cx).is_ready() {
+            if let Some(peer_id) = self.peer_manager.get_disconnected_reserved_peer() {
```

Comment:
Should we push Reconnect events for all disconnected reserved nodes? Otherwise it seems like it could take a while to fully reconnect if we only reconnect to one node per heartbeat interval.

Reply:
good point, I updated the code so that it collects all disconnected peers